app/Http/Controllers/Backend/GeoController.php

```diff
@@ -159,6 +103,74 @@ public static function calculateDistanceBetweenCoordinates(
         return round($distance);
     }
 
+    /**
+     * @throws JsonException
+     */
+    private static function getPolylineBetween(HafasTrip $hafasTrip, TrainStopover $origin, TrainStopover $destination) {
+        $geoJson = self::getPolylineWithTimestamps($hafasTrip);
+
+        $originIndex      = null;
+        $destinationIndex = null;
+        foreach ($geoJson->features as $key => $data) {
+            if (!isset($data->properties->id)) {
+                continue;
+            }
+            if ($originIndex === null
+                && $origin->trainStation->ibnr === (int) $data->properties->id
+                && $origin->departure_planned->is($data->properties->departure_planned) //Important for ring lines!
+            ) {
+                $originIndex = $key;
+            }
+            if ($destinationIndex === null
+                && $destination->trainStation->ibnr === (int) $data->properties->id
+                && $destination->arrival_planned->is($data->properties->arrival_planned) //Important for ring lines!
+            ) {
+                $destinationIndex = $key;
+            }
+        }
+
+        $slicedFeatures    = array_slice($geoJson->features, $originIndex, $destinationIndex - $originIndex + 1);
+        $geoJson->features = $slicedFeatures;
+        return $geoJson;
+    }
+
+    /**
+     * Timestamps in the GeoJSON are required to calculate the distance of ring lines correctly.
+     *
+     * @param HafasTrip $hafasTrip
+     *
+     * @return mixed
+     * @throws JsonException
+     */
+    private static function getPolylineWithTimestamps(HafasTrip $hafasTrip): stdClass {
+        $geoJsonObj = json_decode($hafasTrip->polyline->polyline, false, 512, JSON_THROW_ON_ERROR);
+        $stopovers  = $hafasTrip->stopoversNEW;
+
+        $stopovers = $stopovers->map(function($stopover) {
```

Comment:
added this map here

Reply:
I don't see any relevanc for this map... you don't need `passed` to be set by default?